hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -323,6 +328,35 @@ private synchronized void writeCurrentBufferToService() throws IOException {
     final long offset = position;
     position += bytesLength;
 
+    if (this.isAppendBlob) {
```

Comment:
Could you move this append blob handling to a separate method and call the same from here.

Reply:
done